### PR TITLE
_v3_ [Team Video] explore page + navbar

### PR DESCRIPTION
**Pull request by sofiiagran**
## Description

Created explore page and navbar. 

This merge request is mainly by team video streaming @Mikkelmmc @BenjaminVilladsen @Vagn99 @c3rtificate @mua105310 @carlryskov, the subscribe functionality is created by team offline @sofiiagran. 

## Changes

Created the explore page and changed the navigation bar.
Now user can subscribe from the explore page and is then redirected to the course.

Also made a change to the icon for the category "sewing", as the old icon did not work. 

## Related Issues

PBI: I want to subscribe to the courses I'm interested in
PBI: I want to see all available courses
PBI: I want to filter courses based on category

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

Skærmbillede 2023-10-24 kl  15 32 12
Skærmbillede 2023-10-23 kl  10 35 32
Skærmbillede 2023-10-24 kl  15 32 19
Skærmbillede 2023-10-23 kl  10 35 55

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Works with staging backend.

Use production database and log in with Ola.lola@mail.dk with pass: Password 

## Notes for Reviewers

All storage services and APIs have been tested, so we are sure they are loading correctly.

All utility functions are also tested. 

e2e testing is almost impossible at this point, it will be implemented at some point, however, it is not a priority for now. We believe this PR is ready to be merged into dev and staging for the user tests on Thursday. 

(remember what Luiza/Frederik/Daniel said about valuing technical debt over giving value to stakeholders)

### CODE REVIEW

File: eml/components/explore/FilterNavBar.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import SearchBar from './SearchBar'; 
+import { View, Text, Pressable } from 'react-native';
+import { ScrollView } from 'react-native-gesture-handler';
+
+const categories = [
```
Comment (Mafusn) [inline eml/components/explore/FilterNavBar.jsx:6]: Maybe put this in a seperate file if intended to use in production

File: eml/screens/explore/Explore.jsx [hunk 2]
```diff
@@ -1,87 +1,203 @@
...
+      setIsSubscribed(results);
+    });
+    return update;
+
+  }, [navigation, subCourses]);
+
+  ///---------------------------------------------///
+
+  // Function to dermine category
+  function determineCategory(category) {
```
Comment (Mafusn) [inline eml/screens/explore/Explore.jsx:127]: Why is this function defined again here, when it is defined in utilityfunctions.js

File: eml/screens/explore/Explore.jsx [hunk 3]
```diff
@@ -1,87 +1,203 @@
...
+  const navigation = useNavigation()
+
+
+  /**
+* Determines if the two arrays of courses are different and require an update.
+* @param {Array} courses1 - The first array of courses, typically representing the current state.
+* @param {Array} courses2 - The second array of courses, typically representing the new fetched data.
+* @returns {boolean} - Returns true if the two arrays are different and an update is required, otherwise false.
+*/
+  function shouldUpdate(courses1, courses2) {
```
Comment (Mafusn) [inline eml/screens/explore/Explore.jsx:32]: Why is this function defined again, when it is already defined in utilityFunctions.js

**Review by Mafusn — CHANGES_REQUESTED** [review]
Minor changes

File: eml/App.js [hunk 4]
```diff
@@ -20,19 +20,19 @@ import SectionCompleteScreen from './screens/excercise/SectionCompleteScreen';
 import AsyncStorage from '@react-native-async-storage/async-storage';
```
Comment (Donhusum) [inline eml/App.js:20]: all double quotations " should be single quotation ' in all changed files

File: eml/screens/explore/Explore.jsx (hunk 2 above)
Comment (sofiiagran) [reply eml/screens/explore/Explore.jsx:127]: Because I am stupid, I will fix it lol

File: eml/App.js (hunk 4 above)
Comment (sofiiagran) [reply eml/App.js:20]: Why?

Comment (PrometheusBork) [reply eml/App.js:20]: consistency

**Review by Mafusn — APPROVED** [review]
Looks great!

File: eml/components/navBar/NavBar.jsx [hunk 5]
```diff
@@ -1,15 +1,137 @@
...
+              type="material-community"
+              color={color} // Use the color parameter here
+            />
+          ),
+          tabBarActiveTintColor: 'white', // Set the active text color to white
+          tabBarInactiveTintColor: 'grey', // Set the inactive text color to grey
+        }}
+      />
+
+      {/* <Tab.Screen
```
Comment (Jatewo) [inline eml/components/navBar/NavBar.jsx:98]: Can we delete this?

**Review by Jatewo — CHANGES_REQUESTED** [review]

**Comment by Jatewo** [comment]
Also, the checks are not running

**Review by MortenMunk — CHANGES_REQUESTED** [review]
I can't see checks on this  PR?
Has this PR passed github actions?

**Review by JackSMunn — APPROVED** [review]
Checks are not running, and I have not tested locally. 

Other than that it looks fine. 

I have approved on the basis of some other people have made this branch work locally, and checks are running (which they aren't)

**Review by JackSMunn — CHANGES_REQUESTED** [review]
Checks need to run

**Comment by sofiiagran** [comment]
Github action passed, ready for approval @JackSMunn , @MortenMunk , @Jatewo

**Review by JackSMunn — APPROVED** [review]

**Review by Jatewo — APPROVED** [review]

File: eml/components/explore/FilterNavBar.jsx (hunk 1 above)
Comment (MortenMunk) [reply eml/components/explore/FilterNavBar.jsx:6]: I agree

File: eml/test/components/welcomeScreen.test.js
Comment (MortenMunk) [inline eml/test/components/welcomeScreen.test.js:1]: Is this deleted on purpose?

**Review by MortenMunk — CHANGES_REQUESTED** [review]
Small suggestions:
- Move categories to separate file maybe?,
and a question:
- Is WelcomeSceen test supposed to be deleted?

I cannot run mobile on my computer right now, but that should not hold back this PR as it already has 3 approvals. Just make sure it runs on your system 👍🥇

File: eml/test/components/welcomeScreen.test.js
Comment (PrometheusBork) [reply eml/test/components/welcomeScreen.test.js:1]: Yes, this is a duplicate of another test

**Review by MortenMunk — APPROVED** [review]
Looks fine to me 👍 🥇

**Pull request merged**